pkg/cortex/modules.go

```diff
@@ -569,6 +580,58 @@ func (t *Cortex) stopMemberlistKV() (err error) {
 	return nil
 }
 
+func (t *Cortex) initDataPurger(cfg *Config) (err error) {
+	if !cfg.DataPurgerConfig.EnablePurger {
+		return nil
+	}
+
+	var indexClient chunk.IndexClient
+	indexClient, err = storage.NewIndexClient(cfg.Storage.DeleteStoreConfig.Store, cfg.Storage, cfg.Schema)
+	if err != nil {
+		return
+	}
+
+	var deleteStore *chunk.DeleteStore
+	deleteStore, err = chunk.NewDeleteStore(cfg.Storage.DeleteStoreConfig, indexClient)
+	if err != nil {
+		return
+	}
+
+	var storageClient chunk.ObjectClient
+	storageClient, err = storage.NewStorageClient(cfg.DataPurgerConfig.ObjectStoreType, cfg.Storage)
+	if err != nil {
+		return
+	}
+
+	t.dataPurger, err = purger.NewDataPurger(cfg.DataPurgerConfig, deleteStore, t.store, storageClient)
+	if err != nil {
+		return
+	}
+
+	go t.dataPurger.Run()
+
+	var deleteRequestHandler *purger.DeleteRequestHandler
+	deleteRequestHandler, err = purger.NewDeleteRequestHandler(deleteStore)
+	if err != nil {
+		return
+	}
+
+	subrouter := t.server.HTTP.PathPrefix(cfg.HTTPPrefix).Subrouter()
```

Comment:
This can be done in one line
```suggestion
	subrouter := t.server.HTTP.PathPrefix(cfg.HTTPPrefix+"/api/v1/admin/tsdb").Subrouter()
```